databases/week1/hyf_lesson1.sql

```diff
@@ -0,0 +1,52 @@
+--1-- Find out how many tasks are in the task table
+SELECT count(*)
+FROM task;
+
+--2-- Find out how many tasks in the task table do not have a valid due date
+SELECT count(*)
+FROM task
+WHERE due_date is NULL;
+
+--3-- Find all the tasks that are marked as done
+SELECT *
+from task
+where status_id = 3;
+
+--4-- Find all the tasks that are not marked as done
+SELECT *
+FROM task
+WHERE NOT status_id=3;
```

Comment:
Instead of using the status id, prefer to join the status table and use the name column there.

The reason for this is that it makes it easier to understand, just by looking at your query, what you are trying to do.

If the status table did not have an index on status_id and it was a big table, then you should prefer to do what you did, but in this case it is a very small table, and it doesnt seem like the kind of table that would get big over time

Reply:
Oh okey you are right. I will update my codes. Thanks 👍